### PR TITLE
[IMP] l10n_*: invoice label translate

### PR DESCRIPTION
**Pull request by malb-odoo**
Before this PR, the field "invoice_label" was not translatable and since we wanted to have all localisation in english and then translated back in the desired language, the "invoice_label" field was not consistent for most of them.
In previous pr (https://github.com/odoo/odoo/pull/124520) i've translated back the invoice label in the country language to keep the consistency that we lost but in master, i've made the field translatable and added translation.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_tax.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ class AccountTax(models.Model):
         help="The sequence field is used to define order in which the tax lines are applied.")
     amount = fields.Float(required=True, digits=(16, 4), default=0.0)
     description = fields.Char(string='Description', translate=True)
-    invoice_label = fields.Char(string='Label on Invoices')
+    invoice_label = fields.Char(string='Label on Invoices', translate=True)
```
Comment (oco-odoo) [inline addons/account/models/account_tax.py:116]: Doesn't this change require a migration script ? (not sure; maybe the ORM does that efficiently on its own => to check with upgrade)

Comment (william-andre) [reply addons/account/models/account_tax.py:116]: It is managed automatically
https://github.com/odoo/odoo/blob/9e228b6feb09b1afac5e7af6502b834e4755ba15/odoo/fields.py#L1668-L1673
https://github.com/odoo/odoo/blob/9e228b6feb09b1afac5e7af6502b834e4755ba15/odoo/tools/sql.py#L152-L159

**Comment by william-andre** [comment]
@malb-odoo Please resolve conflict

**Review by oco-odoo — APPROVED** [review]
@robodoo r+

**Comment by oco-odoo** [comment]
@robodoo rebase-ff

**Comment by oco-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**